overlord/ifacestate/handlers.go

```diff
@@ -521,10 +524,15 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 		}
 	}
 
-	conn := interfaces.ConnRef{PlugRef: plugRef, SlotRef: slotRef}
-	delete(conns, conn.ID())
-
+	cref := interfaces.ConnRef{PlugRef: plugRef, SlotRef: slotRef}
+	if conn, ok := conns[cref.ID()]; ok && conn.Auto {
+		conn.Undesired = true
```

Comment:
Can you please add a comment here:
`// TODO: remove any stored connection specific state like interface attributes.`

I would also drop the `auto` flag at this stage. What do you think?

Reply:
I'm not sure about setting auto=false (I agree about dropping attributes as a TODO).
If we drop this we will never know it was an auto-connection before, there is no way for it to ever be restored (manual connect will just remove 'undesired' flag). For example, policy check on autoconnect is different from the regular policy check for connect.

Reply:
@stolowski has good points.